runtime/lua/vim/lsp/util.lua

```diff
@@ -587,31 +607,32 @@ function M.fancy_floating_markdown(contents, opts)
       end
     end
   end
-  local width = 0
+  -- Clean input and add padding
   for i, v in ipairs(stripped) do
     v = v:gsub("\r", "")
     if pad_left then v = (" "):rep(pad_left)..v end
     if pad_right then v = v..(" "):rep(pad_right) end
```

Comment:
We can do this in one `string.format` which is better than concats I think.

```lua
left_padding = " ":rep(pad_left)
right_padding = " ":rep(pad_right)

for loop here
 v = string.format('%s%s%s', right_padding, v:gsub("\r", ""), right_padding)
end
```

psuedo code above will not work exactly as written ;)